lib/command/definitions.js

```diff
@@ -2,387 +2,160 @@ const getConfig = require('./utils').getConfig;
 const getTestRoot = require('./utils').getTestRoot;
 const Codecept = require('../codecept');
 const container = require('../container');
-const methodsOfObject = require('../utils').methodsOfObject;
 const output = require('../output');
-const { toTypeDef } = require('../parser');
 const fs = require('fs');
 const path = require('path');
 
-const template = `
-type ICodeceptCallback = (i?: CodeceptJS.{{I}}, current?:any{{callbackParams}}) => void;
+const template = ({
+  helperNames, supportObject, importPaths, translations, hasCustomStepsFile,
+}) => `/// <reference types='codeceptjs' />
+${importPaths.join('\n')}
 
-declare class FeatureConfig {
-  retry(times: number): FeatureConfig
-  timeout(seconds: number): FeatureConfig
-  config(config: object): FeatureConfig
-  config(helperName: string, config: object): FeatureConfig
-}
-
-declare class ScenarioConfig {
-  throws(err: any): ScenarioConfig;
-  fails(): ScenarioConfig;
-  retry(times: number): ScenarioConfig
-  timeout(timeout: number): ScenarioConfig
-  inject(inject: object): ScenarioConfig
-  config(config: object): ScenarioConfig
-  config(helperName: string, config: object): ScenarioConfig
-  injectDependencies(dependencies: { [key: string]: any }): ScenarioConfig
-}
-
-interface ILocator {
-  id?: string;
-  xpath?: string;
-  css?: string;
-  name?: string;
-  frame?: string;
-  android?: string;
-  ios?: string;
-}
+type translation = ${JSON.stringify(translations.vocabulary, null, 2)};
+type WithTranslation<T> = T & import('codeceptjs/typings/utils').Translate<T, translation["actions"]>;
+type methods = ${helperNames.length > 0 ? `${helperNames.join(' & ')} & ` : ''} import('codeceptjs/typings/utils').ActorStatic;
 
-type LocatorOrString = string | ILocator | Locator;
+type ICodeceptCallback = (${convertObjectToParameterList(supportObject)}) => void;
```

Comment:
This callback has a hard order of arguments. And CodeceptJS allows you to transfer here PageObjects in an arbitrary order. And when I don't want to use I but only PageObject, autocomplete suggests methods for I

Reply:
Unfortunately, it is impossible to recreate how codeceptjs works. Right know `Scenario`, `xScenario`, `BeforeSuite`, `AfterSuite`, `Background`, `Before` and `After` have an default order that is the same as in the past. But it can be overwritten within Typescript, e.g.
```ts
Scenario<[MyHelper, CodeceptJS.I]>("...", async (myHelper, I) => {
 ...
});
```